### ISSUE TITLE
Keycloak X - java.lang.ClassNotFoundException

### ISSUE THREAD

**Issue by fidelix** [daniel-frak/keycloak-user-migration#50]
I tested in Keycloak 16.1 and 15.1, with the X distribution:
```
keycloak_1  | 2021-12-30 13:03:08,821 WARN  [org.key.services] (executor-thread-66) KC-SERVICES0013: Failed authentication: java.lang.RuntimeException: java.lang.ClassNotFoundException: org.glassfish.jersey.client.JerseyClientBuilder
keycloak_1  |   at javax.ws.rs.client.ClientBuilder.newBuilder(ClientBuilder.java:102)
keycloak_1  |   at javax.ws.rs.client.ClientBuilder.newClient(ClientBuilder.java:113)
keycloak_1  |   at com.danielfrak.code.keycloak.providers.rest.LegacyProviderFactory.create(LegacyProviderFactory.java:25)
keycloak_1  |   at com.danielfrak.code.keycloak.providers.rest.LegacyProviderFactory.create(LegacyProviderFactory.java:15)
keycloak_1  |   at org.keycloak.storage.AbstractStorageManager.getStorageProviderInstance(AbstractStorageManager.java:229)
keycloak_1  |   at org.keycloak.storage.AbstractStorageManager.lambda$getEnabledStorageProviders$0(AbstractStorageManager.java:98)
keycloak_1  |   at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
keycloak_1  |   at java.base/java.util.stream.SortedOps$RefSortingSink.end(SortedOps.java:400)
keycloak_1  |   at java.base/java.util.stream.Sink$ChainedReference.end(Sink.java:258)
........
.....
keycloak_1  | Caused by: java.lang.ClassNotFoundException: org.glassfish.jersey.client.JerseyClientBuilder
keycloak_1  |   ... 77 more
keycloak_1  |
keycloak_1  | 2021-12-30 13:03:08,829 WARN  [org.key.events] (executor-thread-66) type=LOGIN_ERROR, realmId=local, clientId=zzz-web, userId=null, ipAddress=172.28.0.1, error=invalid_user_credentials, auth_method=openid-connect, auth_type=code, redirect_uri=https://zzz.kc.l/, code_id=bde1ebeb-b2ab-4b2a-9c1a-cc201128f2f1, username=zzzzzz, authSessionParentId=bde1ebeb-b2ab-4b2a-9c1a-cc201128f2f1, authSessionTabId=uQEeyvaPdhM

```

**Comment by daniel-frak** [comment]
Unfortunately, at this time I'm not sure what might be causing this.

As a random check, does changing `resteasy.version` in the POM to `4.5.9.Final` help?

**Comment by manizm** [comment]
This is most probably due to the fact that keycloak has bumped `resteasy` core version to `4.7.5` and it has breaking changes. Mainly, `ResteasyClientBuilder` is moved from `rest-client` to `resteasy-client-api`.

So, adding this dependency should resolve the issue

**Comment by rfeijolo** [comment]
> This is most probably due to the fact that keycloak has bumped `resteasy` core version to `4.7.5` and it has breaking changes. Mainly, `ResteasyClientBuilder` is moved from `rest-client` to `resteasy-client-api`.
> 
> And you can't use the `3x` version due to conflict with keycloak and quarkus.
> 
> So, adding this dependency should resolve the issue.

Just tested this approach with Keycloak 17.0.1 and unfortunately it didn't work.
---
I dropped reasteasy in favor of apache http client and it worked. I can send a Pull Request next week 😉
[This issue](https://github.com/keycloak/keycloak/issues/10555#issuecomment-1064217841) is related and explains why this error is happening, basically they are reducing the dependency graph of the server and some dependencies are gone.

**Comment by daniel-frak** [comment]
@rfeijolo normally I would be against creating new dependencies when we can technically use existing ones but from the thread you linked to it seems that the Keycloak devs themselves are suggesting this...

There seem to be problems with the RestEasy version every time the Keycloak version gets bumped, as well, so this could kill two birds with one stone.

I would be grateful if you could send the Pull Request :)

**Comment by rfeijolo** [comment]
The PR that fixes the issue is open PR #52, tell me what you think @daniel-frak.
Just tested and it seems to work with the current version of Keycloak (17.0.1).

**Comment by daniel-frak** [comment]
Since PR #52 has been merged, the bug should be fixed and so I'm closing this issue.

Let me know if any problems persist and I'll reopen the issue.
